lib/floki/selector.ex

```diff
@@ -143,8 +158,24 @@ defmodule Floki.Selector do
 
   defp do_classes_matches?(nil, _classes), do: false
 
+  defp do_classes_matches?(class_attr_value, [class | _])
+       when bit_size(class_attr_value) < bit_size(class) do
+    false
+  end
+
+  defp do_classes_matches?(class_attr_value, [class])
+       when bit_size(class_attr_value) == bit_size(class) do
+    class == class_attr_value
+  end
+
+  defp do_classes_matches?(class_attr_value, [class]) do
+    Enum.member?(String.split(class_attr_value, [" ", "\t", "\n"]), class)
```

Comment:
I think this is not exactly the same from the regex version, because the occurrence of multiple whitespace chars in sequence are treated differently.

For example, if we have a class attribute that contains multiple spaces between classes values, the former version would capture that correctly, while the new version does not:

```elixir
class_attr_value = "my class\tanother   one"

# new
String.split(class_attr_value, [" ", "\t", "\n"])
#=> ["my", "class", "another", "", "", "one"]

# old
String.split(class_attr_value, ~r/\s+/)
#=> ["my", "class", "another", "one"]
```

We can, however, reject the empty binaries from the match. Can you try this and see if there is an impact in performance?

Reply:
Done, fa64b01261028d47eb5d41c1f6aab840a781f636.
Adding `trim: true` didn't had performance impact in the benchmark, but it did reduce the memory usage, since we won't have this empty strings in the list.